lib/engine/connection.rb

```diff
@@ -20,7 +20,6 @@ def self.connect!(hex)
           connection = connection.branch!(path)
           next if connection.paths.include?(path)
           next if (connection.nodes & path.nodes).any?
-          next if connection.paths.any? { |p| p.hex == hex && (p.exits & path.exits).any? }
```

Comment:
was this removed due to lanes? will this cause issues?

Reply:
Two reasons:
1. It's redundant given the new way branch! has been coded
2. It doesn't handle lanes

I initially changed it to handle lanes and realized it looked exactly like what is now in branch!